tree.go

```diff
@@ -54,10 +62,14 @@ type tNumberType struct {
 	scale int
 }
 
-type tSliceType struct {
+type SliceType struct {
 	elementType Type
 }
 
+func (s *SliceType) ElementType() Type {
```

Comment:
Same comment as above, because this is a type related helper, it should live in `types.go`